oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java

```diff
@@ -308,6 +310,9 @@ private void createSecondaryStore(LeaseCheckMode leaseCheckNode, boolean withFai
                 .setLeaseCheckMode(leaseCheckNode)
                 .setDocumentStore(ds2).setAsyncDelay(0);
         store2 = documentMKBuilder2.getNodeStore();
+        // custom fullGcMode needs to be set after each node store creation, since the VersionGarbageCollector
+        // constructor sets the fullGcMode to the value read from OSGI Configuration - method setFullGcMode
+        writeStaticField(VersionGarbageCollector.class, "fullGcMode", fullGcMode, true);
```

Comment:
can't we use `setFullGcMode` here?

Reply:
ok, I see, we had it like this previously .. I'd still vote to introducing a dedicated testing-only setter (`static void setFullGcMode(FullGCMode)`) instead of reflection (for this and a few other similar cases for fullgcmode in this and the other test class) - makes code maintenance easier imv

Reply:
I propose we make that change when we do a larger scale refactoring. Changing the way we set fullGCMode in unit tests now would also require updating all of the related unit tests (for consistency), and there are a lot of them.

We should do refactoring in a separate, dedicated ticket, not as a part of this feature, IMHO.